src/index.js

```diff
@@ -1,23 +1,45 @@
 import React from "react";
 import ReactDOM from "react-dom";
 import { Provider } from "react-redux";
-import { createStore } from "redux";
+import { applyMiddleware, combineReducers, createStore } from "redux";
+import thunk from "redux-thunk";
+import { composeWithDevTools } from "redux-devtools-extension";
 
 import "./scss/index.scss";
 
 import App from "./App";
 import rootReducer from "./reducers/root";
 import * as serviceWorker from "./serviceWorker";
 
-import { loginWithBakery } from "./juju";
+import { fetchModelList, loginWithBakery } from "./juju";
+import jujuReducers from "./juju/reducers";
 
-loginWithBakery();
+const reduxStore = createStore(
+  combineReducers({
+    root: rootReducer,
+    juju: jujuReducers
+  }),
+  composeWithDevTools(applyMiddleware(thunk))
+);
+
+async function connectAndListModels(reduxStore) {
+  try {
+    const conn = await loginWithBakery();
+    reduxStore.dispatch(fetchModelList(conn));
```

Comment:
`dispatch` is typically called with an action creator which returns a standard flux action. It is a bit unusual to dispatch a regular function here which in turn dispatches another redux action. Could you instead dispatch an action from fetchModelList which performs the API call in middleware or in the action creator, and then responds with a SUCCESS action of some sort (settings some state so the UI knows the fetch has completed).